app/models/plan.rb

```diff
@@ -359,9 +362,15 @@ def renewal_plan=(new_renewal_plan)
     end
   end
 
-  def renewal_plan
+  def renewal_plan(date = nil)
     return @renewal_plan if defined? @renewal_plan
-    @renewal_plan = Plan.find(renewal_plan_id) unless renewal_plan_id.blank?
+
+    if date.present? && renewal_plan_mappings.by_date(date.year).present?
```

Comment:
Please pass the date instead of year....ex. if we're renewing for 5/1/2018, we need to pass this date to by_date scope.